libs/community/langchain_community/vectorstores/opensearch_vector_search.py

```diff
@@ -733,6 +793,115 @@ async def adelete(
             item.get("delete", {}).get("error") for item in response["items"]
         )
 
+    def configure_search_pipelines(
+        self,
+        pipeline_name: str,
+        keyword_weight: float = 0.7,
+        vector_weight: float = 0.3,
+    ) -> dict:
+        """
+        Configures a search pipeline for hybrid search.
+        Args:
+            pipeline_name: Name of the pipeline
+            keyword_weight: Weight for keyword search
+            vector_weight: Weight for vector search
+        Returns:
+            response: Acknowledgement of the pipeline creation.
+            (if there is any error while configuring the pipeline, it will return None)
+        Raises:
+            Exception: If an error occurs
+        """
+        path = f"/_search/pipeline/{pipeline_name}"
+
+        payload = {
+            "description": "Post processor for hybrid search",
+            "phase_results_processors": [
+                {
+                    "normalization-processor": {
+                        "normalization": {"technique": "min_max"},
+                        "combination": {
+                            "technique": "arithmetic_mean",
+                            "parameters": {"weights": [keyword_weight, vector_weight]},
+                        },
+                    }
+                }
+            ],
+        }
+        
+        response = self.client.transport.perform_request(
+            method="PUT", url=path, body=payload
+        )
+        return response
+        
+
+    def search_pipeline_exists(self, pipeline_name: str) -> bool:
+        """
+        Checks if a search pipeline exists.
+
+        Args:
+            pipeline_name: Name of the pipeline
+
+        Returns:
+            bool: True if the pipeline exists, False otherwise
+
+        Raises:
+            Exception: If an error occurs
+
+        Example:
+            >>> search_pipeline_exists("my_pipeline_1")
+            True
+            >>> search_pipeline_exists("my_pipeline_2")
+            False
+        """
+        
+        existed_pipelines = self.client.transport.perform_request(
+            method="GET", url=f"/_search/pipeline/"
+        )
+
+        return pipeline_name in existed_pipelines
+
+    def get_search_pipeline_info(self, pipeline_name: str) -> Optional[Dict]:
```

Comment:
sorry just caught this: `pipeline_name` needs to be sanitized here potentially -- i don't know what the underlying client being used, but there's a chance that escaping will be possible.
You could verify that pipeline_name is a proper identifier (single if statement)

```
In [6]: "..qwe".isidentifier()   
```